bundles/CoreBundle/Command/MaintenanceCommand.php

```diff
@@ -80,8 +86,33 @@ protected function execute(InputInterface $input, OutputInterface $output)
     {
         $validJobs = $this->getArrayOptionValue($input, 'job');
         $excludedJobs = $this->getArrayOptionValue($input, 'excludedJobs');
-
-        $this->maintenanceExecutor->executeMaintenance($validJobs, $excludedJobs, (bool) $input->getOption('force'));
+        $async = (bool)$input->getOption('async');
+
+        $this->maintenanceExecutor->executeMaintenance(
+            $validJobs,
+            $excludedJobs,
+            (bool)$input->getOption('force'),
+            $async
+        );
+
+        if (!$async) {
+            trigger_deprecation(
+                'pimcore/pimcore',
+                '10.1',
```

Comment:
If this is for 10.1, then we need to change the base branch to 10.1 😉

Reply:
This is more of a new feature than a bugfix, no? So it should go into the next 10.x

Reply:
Absolutely :)